### PR TITLE
Feat/django 42 compatible

### PR DESCRIPTION
**Pull request by vipulnarang95**
Added Django support 4.2 and Python 3.10
Dropped support for django<3.2 and python<3.8

### CODE REVIEW

**Review by Aiky30 — COMMENTED** [review]
Hi @vipulnarang95,

This is a FIL specific package and not a DCA supported package so I can only offer a favour personally here. 

The test failures look to be something django admin related. To debug this I would recommend adding breakpoints in the test and tracing the code through to the point of failure to see what is actually happening underneath.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,9 @@ Changelog
 
 Unreleased
 ==========
+* Python 3.10 support added
+* Python 3.7 support removed
+* Django 4.2 support added
```
[the remark this answers is not in the document]
Comment (vipulnarang95) [reply CHANGELOG.rst:10]: Ok

**Comment by vipulnarang95** [comment]
Hi @Aiky30, Thankyou for the favor. Much appreciated.. I fixed major of them
**Only one test is failing now.** Can you please suggest something on this?
FAIL: test_menuitem_changelist_custom_js (tests.test_admin.MenuItemAdminChangeListViewTestCase)
Check that custom javascript for overriding treebeard js is present.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/vipul/Desktop/django/djangocms-navigation/tests/test_admin.py", line 1162, in test_menuitem_changelist_custom_js
    self.assertContains(
  File "/Users/vipul/Desktop/django/djangocms-navigation/venv/lib/python3.10/site-packages/django/test/testcases.py", line 660, in assertContains
    self.assertTrue(
AssertionError: False is not true : Couldn't find 'djangocms_navigation/js/navigation-tree-admin.js' in response

File: tests/test_admin.py [hunk 2]
```diff
@@ -29,14 +28,15 @@
...
+from djangocms_navigation.compat import DJANGO_4_2
 from djangocms_navigation.models import Menu, MenuContent, MenuItem
 from djangocms_navigation.test_utils import factories
 
 from .utils import UsefulAssertsMixin, disable_versioning_for_navigation
 
 
-version = list(map(int, django.__version__.split('.')))
-GTE_DJ21 = version[0] >= 2 and version[1] >= 1
+if not DJANGO_4_2:
```
[the remark this answers is not in the document]
Comment (vipulnarang95) [reply tests/test_admin.py:38]: ok

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -16,9 +16,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [ 3.7, 3.8, 3.9 ]
+        python-version: [ 3.8, 3.9, '3.10' ]
```
[the remark this answers is not in the document]
Comment (vipulnarang95) [reply .github/workflows/test.yml:19]: Agreed but we have updated all Fil specific branches with same code and we would be upgrading to CMS4.1 just after this upgrade.

**Comment by vipulnarang95** [comment]
Hi @Aiky30 , Need your favor in this. The test which I mentioned above is failing as the treebeard template tag is not working as expected and the custom js is not find in the response. Would you be able to suggest something?

**Comment by joshyu** [comment]
@vipulnarang95 , I have fixed your errors, please check my PR : https://github.com/vipulnarang95/djangocms-navigation/pull/2 

@Aiky30 , please also help to review my solution.

File: tests/requirements/requirements_base.txt [hunk 4]
```diff
@@ -1,15 +1,19 @@
...
-https://github.com/django-cms/djangocms-moderation/tarball/master#egg=djangocms-moderation
-https://github.com/FidelityInternational/djangocms-references/tarball/1.4.3#egg=djangocms-references
-https://github.com/django-cms/djangocms-alias/tarball/1.11.0#egg=djangocms-alias
+
+# Unreleased django-cms 4.0 compatible packages
+https://github.com/django-cms/django-cms/tarball/release/4.0.1.x#egg=django-cms
+https://github.com/divio/djangocms-text-ckeditor/tarball/master#egg=djangocms-text-ckeditor
+https://github.com/django-cms/djangocms-versioning/tarball/support/django-cms-4.0.x#egg=djangocms-versioning
+https://github.com/joshyu/djangocms-version-locking/tarball/feat/django-42-compatible#egg=djangocms-version-locking
+https://github.com/FidelityInternational/djangocms-moderation/tarball/feature/django-42-compat#egg=djangocms-moderation
```
Comment (joshyu) [inline tests/requirements/requirements_base.txt:17]: replace `djangocms-moderation` to use  https://github.com/django-cms/djangocms-moderation/tarball/master#egg=djangocms-moderation

**Pull request merged**